lang/funcs/collection.go

```diff
@@ -440,8 +440,7 @@ var FlattenFunc = function.New(&function.Spec{
 			return cty.NilType, fmt.Errorf("can only flatten lists, sets and tuples")
 		}
 
-		outputList := make([]cty.Value, 0)
-		retVal := flattener(outputList, args[0])
+		retVal, _ := flattener(args[0])
```

Comment:
I think we would need to check `known` here too, and return `cty.DynamicPseudoType` if it's true... otherwise the final result might not be compatible with the type that was predicted, causing an error.

Reply:
oh yes! I put it in the wrong place, then removed it wondering why it was there :/